File: benchmarks/cpp/sort_n.cpp
#include <iostream>
#include <chrono>
#include <vector>
#include <algorithm>
#include <cstdlib>
long long f(long long n) {
    std::vector<long long> arr(n);
    for (long long i = 0; i < n; i++) arr[i] = n - i;
    std::sort(arr.begin(), arr.end());
    return arr[0];
}
int main(int argc, char** argv) {
    long long n = std::atoll(argv[1]);
    auto t0 = std::chrono::high_resolution_clock::now();
    volatile long long r = f(n);
    auto t1 = std::chrono::high_resolution_clock::now();
    std::chrono::duration<double, std::milli> ms = t1 - t0;
    std::cout << "ELAPSED_MS=" << ms.count() << " RESULT=" << r << std::endl;
    return 0;
}

File: benchmarks/java/SortN.java
import java.util.Arrays;
public class SortN {
    static long f(long n) {
        int size = (int) n;
        long[] arr = new long[size];
        for (int i = 0; i < size; i++) arr[i] = size - i;
        Arrays.sort(arr);
        return arr[0];
    }
    public static void main(String[] args) {
        long n = Long.parseLong(args[0]);
        long t0 = System.nanoTime();
        long r = f(n);
        long t1 = System.nanoTime();
        System.out.println("ELAPSED_MS=" + (t1 - t0) / 1_000_000.0 + " RESULT=" + r);
    }
}
